src/parallel/DomainDecompMPIBase.h

```diff
@@ -106,7 +107,31 @@ class DomainDecompMPIBase: public DomainDecompBase {
 	}
 
 	void collCommAllreduceSumAllowPrevious() override {
+	    // TODO: Write Timer in here; or in DomainDecompBase
+#if defined(ENABLE_MPI)
+        Timer mpi_ReductionAll_timer;
+	    mpi_ReductionAll_timer.start();
+#endif
 		_collCommunication->allreduceSumAllowPrevious();
+#if defined(ENABLE_MPI)
+	    mpi_ReductionAll_timer.stop();
+	    double runtime = mpi_ReductionAll_timer.get_etime();
+
+	    MPI_Comm_size(_comm, &size );
+        MPI_Comm_rank(_comm, &rank );
+
+        _processTimer.insertTime(int(rank), double(runtime));
```

Comment:
please use c++11 casts instead of c-style casts, i.e.,
```suggestion
        _processTimer.insertTime(static_cast<int>(rank), runtime);
```
the second cast is also not needed.